src/topology/sheaves/locally_surjective.lean

```diff
@@ -0,0 +1,116 @@
+/-
+Copyright (c) 2022 Sam van Gool and Jake Levinson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sam van Gool, Jake Levinson
+-/
+
+import topology.sheaves.presheaf
+import topology.sheaves.stalks
+
+/-!
+
+Locally surjective maps of presheaves.
+
+Let X be a topological space, ℱ and 𝒢 presheaves on X, T : ℱ ⟶ 𝒢 a map.
+
+In this file we formulate two notions for what it means for
+T to be locally surjective:
+
+  1. For each open set U, each section t ∈ 𝒢(U) is in the image of T
+     after passing to some open cover of U.
+
+  2. For each x : X, the map of *stalks* Tₓ : ℱₓ ⟶ 𝒢ₓ is surjective.
+
+We prove that these are equivalent.
+
+-/
+
+universes v u
+
+noncomputable theory
+
+open category_theory
+open topological_space
+open opposite
+
+namespace Top.presheaf
+
+section locally_surjective
+
+local attribute [instance] concrete_category.has_coe_to_fun
+local attribute [instance] concrete_category.has_coe_to_sort
+
+/-- Let C be a concrete category, X a topological space. -/
+variables {C : Type u} [category.{v} C] [concrete_category.{v} C] {X : Top.{v}}
+
+/-- Let ℱ, 𝒢 : (opens X)ᵒᵖ ⥤ C be C-valued presheaves on X. -/
+variables {ℱ 𝒢 : X.presheaf C}
+
+/-- A map of presheaves T : ℱ ⟶ 𝒢 is **locally surjective** if for
+any open set U, section t over U, and x ∈ U, there exists an open set
+x ∈ V ⊆ U such that $T_*(s_V) = t|_V$. -/
+def is_locally_surjective (T : ℱ ⟶ 𝒢) :=
+  ∀ (U : opens X) (t : 𝒢.obj (op U)) (x : X) (hx : x ∈ U),
+  ∃ (V : opens X) (ι : V ⟶ U) (hxV : x ∈ V) (s : ℱ.obj (op V)),
+  T.app _ s = 𝒢.map ι.op t
+
+section surjective_on_stalks
+
+variables
+  [category_theory.limits.has_colimits C]
+  [category_theory.limits.preserves_filtered_colimits (forget C)]
+
+/-- An equivalent condition for a map of presheaves to be locally surjective
+is for all the induced maps on stalks to be surjective. -/
+lemma locally_surjective_iff_surjective_on_stalks (T : ℱ ⟶ 𝒢) :
+  is_locally_surjective T ↔
+  ∀ (x : X), function.surjective ((stalk_functor C x).map T) :=
+begin
+  split; intro hT,
+  { /- human proof:
+    Let g ∈ Γₛₜ 𝒢 x be a germ. Represent it on an open set U ⊆ X
+    as ⟨t, U⟩. By local surjectivity, pass to a smaller open set V
+    on which there exists s ∈ Γ_ ℱ V mapping to t |_ V.
+    Then the germ of s maps to g -/
+
+    -- Let g ∈ Γₛₜ 𝒢 x be a germ.
+    intros x g,
+    -- Represent it on an open set U ⊆ X as ⟨t, U⟩.
+    rcases 𝒢.germ_exist x g with ⟨U, hxU, t, rfl⟩,
```

Comment:
You can also write it as follows, which is more inline with what you are doing mathematically IMO.
```suggestion
    obtain ⟨U, hxU, t, rfl⟩ :=  𝒢.germ_exist x g,
```